lib/rspec/core/reporter.rb

```diff
@@ -2,6 +2,15 @@ module RSpec::Core
   # A reporter will send notifications to listeners, usually formatters for the
   # spec suite run.
   class Reporter
+    # @private
+    NOTIFICATIONS =
+      [
+        :close, :deprecation, :deprecation_summary, :dump_failures, :dump_pending,
+        :dump_profile, :dump_summary, :example_failed, :example_group_finished,
+        :example_group_started, :example_passed, :example_pending, :example_started,
+        :message, :seed, :start, :start_dump, :stop
+      ]
```

Comment:
I think this should be a set rather than an array, so that `NOTIFICATIONS.include?` has constant-time complexity (given that's how you use it).